test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -127,39 +134,47 @@ impl MessageTester {
         let test_result_coordinator_url = settings.trc_url();
         let reporting_client = TrcClient::new(test_result_coordinator_url);
 
-        let message_handler: Box<dyn MessageHandler + Send> = match settings.test_scenario() {
-            TestScenario::Initiate => {
-                let batch_id = settings
-                    .batch_id()
-                    .ok_or(MessageTesterError::MissingBatchId)?;
-                Box::new(ReportResultMessageHandler::new(
+        let message_handler: Option<Box<dyn MessageHandler + Send>> = match settings.test_scenario()
```

Comment:
We won't always have a message handler now (i.e. Initiate mode), so needs to be an option.